### PR TITLE
Webapp snapshot and undelete commands with mock tests

### PR DESCRIPTION
**Pull request by Nking92**
Adds webapp snapshot commands. Adds mock tests for snapshots and deleted webapp restore commands since live testing is not possible due to limitations in how the features work.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py [hunk 1]
```diff
@@ -301,6 +301,17 @@ def load_arguments(self, _):
         c.argument('overwrite', help='Overwrite the source webapp, if --target-name is not specified', action='store_true')
         c.argument('ignore_hostname_conflict', help='Ignores custom hostnames stored in the backup', action='store_true')
 
+    with self.argument_context('webapp config snapshot') as c:
+        c.argument('name', arg_type=webapp_name_arg_type)
+        c.argument('slot', options_list=['--slot', '-s'], help='The name of the slot.')
+
+    with self.argument_context('webapp config snapshot restore') as c:
+        c.argument('time', help='Timestamp of the snapshot to restore.')
+        c.argument('restore_config', help='Restore the previous configuration along with web app content.')
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py]: please provide sample values

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py [hunk 2]
```diff
@@ -301,6 +301,17 @@ def load_arguments(self, _):
         c.argument('overwrite', help='Overwrite the source webapp, if --target-name is not specified', action='store_true')
         c.argument('ignore_hostname_conflict', help='Ignores custom hostnames stored in the backup', action='store_true')
 
+    with self.argument_context('webapp config snapshot') as c:
+        c.argument('name', arg_type=webapp_name_arg_type)
+        c.argument('slot', options_list=['--slot', '-s'], help='The name of the slot.')
+
+    with self.argument_context('webapp config snapshot restore') as c:
+        c.argument('time', help='Timestamp of the snapshot to restore.')
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py]: please provide sample values

**Comment by Nking92** [comment]
@tjprescott is this the error that's failing the CI?

************* Module azure.cli.command_modules.appservice._help
C:  1, 0: Too many lines in module (1019/1000) (too-many-lines)

How should I fix this? It looks like the help messages for the new commands pushed the help file over the line limit.

**Comment by tjprescott** [comment]
@Nking92 you can simply put this near the top of the file

```Python
# pylint: disable=too-many-lines
```

**Comment by Nking92** [comment]
I'm ready to merge if it looks good to you. No hard deadline for this one, but would like to get it done as soon as we can.

**Pull request merged**